nntrainer/layers/preprocess_flip_layer.cpp

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	preprocess_flip_layer.cpp
+ * @date	20 January 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Preprocess Random Flip Layer Class for Neural Network
+ *
+ */
+
+#include <random>
+
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <preprocess_flip_layer.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+const std::string PreprocessFlipLayer::type = "preprocess_flip";
+
+const std::string PreprocessFlipLayer::flip_horizontal = "horizontal";
+const std::string PreprocessFlipLayer::flip_vertical = "vertical";
+const std::string PreprocessFlipLayer::flip_horizontal_vertical =
+  "horizontal_and_vertical";
+
+int PreprocessFlipLayer::initialize(Manager &manager) {
+  output_dim = input_dim;
+
+  rng.seed(getSeed());
+  flip_dist = std::uniform_real_distribution<float>(0.0, 1.0);
+
+  return ML_ERROR_NONE;
+}
+
+void PreprocessFlipLayer::setProperty(const PropertyType type,
+                                      const std::string &value) {
+  switch (type) {
+  case PropertyType::flip_direction:
+    if (!value.empty()) {
+      if (istrequal(value, flip_horizontal)) {
+        flipdirection = FlipDirection::horizontal;
+      } else if (istrequal(value, flip_vertical)) {
+        flipdirection = FlipDirection::vertical;
+      } else if (istrequal(value, flip_horizontal_vertical)) {
+        flipdirection = FlipDirection::horizontal_and_vertical;
+      } else {
+        throw std::invalid_argument("Argument flip direction is invalid");
+      }
+    }
+    break;
+  default:
+    Layer::setProperty(type, value);
+    break;
+  }
+}
+
+void PreprocessFlipLayer::forwarding() {
```

Comment:
Could you check if flipdirection is used in the function?

Reply:
oops.
Added.